### PR TITLE
Types: overload put-effect for use thunk actions

### PR DESCRIPTION
**Pull request by eiire**
| Q                        | A  |
| ------------------------ | ---  |
| Fixed Issue            | `https://github.com/redux-saga/redux-saga/issues/2271`  |

Typescript error when dispatching action created by createAsyncThunk.

### CODE REVIEW

File: packages/core/types/ts3.6/effects.d.ts [hunk 1]
```diff
@@ -381,22 +381,36 @@ export type HelperWorkerParameters<T, Fn extends (...args: any[]) => any> = Last
...
  *
- * @param action [see Redux `dispatch` documentation for complete info](https://redux.js.org/api/store#dispatchaction)
+ * @param action [see Redux `dispatch` documentation for complete info](http://redux.js.org/docs/api/Store.html#dispatch)
  */
 export function put<A extends Action>(action: A): PutEffect<A>
 
+/**
+ * Interface for using thunk action from Redux Toolkit in `put`.
+ */
+interface AsyncThunkAction {}
```
Comment (Andarist) [inline packages/core/types/ts3.6/effects.d.ts:391]: This is an empty interface - this effectively behaves like `any`, see the playground here:
https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgIIGcCeIEBUAWAriANaoJjAD2IyA3gL4BQTAJhAgDZxQozEVqtAA6EwAHlTIIAD0ghW6NFhwFiZQTQB8ACjBFS5SjQBcaAJRmAblWCsWAegfIAAmHQBaWcI5gvUKCooJlEwHQBGACYAZnMWIA

**Comment by neurosnap** [comment]
@eiire Greetings!  Are you planning on picking this back up?

**Comment by eiire** [comment]
> @eiire Greetings! Are you planning on picking this back up?

Hi! I'm sorry, but not in the near future

**Comment by neurosnap** [comment]
Please feel free to pick this back up anytime and I'll try to help.

**Pull request closed without merging**